hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -98,7 +97,12 @@ class IncrementalRelation(val sqlContext: SQLContext,
 
   override def schema: StructType = usedSchema
 
-  override def buildScan(): RDD[Row] = {
+  override def buildScan(requiredColumns: Array[String]): RDD[Row] = {
+    log.info(s"buildScan requiredColumns = ${requiredColumns.mkString(",")}")
+    sqlContext.sparkSession.sessionState.conf.setConfString("spark.sql.parquet.filterPushdown", "true")
+    sqlContext.sparkSession.sessionState.conf.setConfString("spark.sql.parquet.recordLevelFilter.enabled", "true")
+    sqlContext.sparkSession.sessionState.conf.setConfString("spark.sql.parquet.enableVectorizedReader", "false")
```

Comment:
the reason set to false is that the verctorizedReader is not supported for at the moment for IncrementalRelation?

Reply:
It is because "spark.sql.parquet.enableVectorizedReader" and "spark.sql.parquet.recordLevelFilter.enabled" cannot be true at the same time according to the Parquet documentation. We have to disable verctorizedReader if we want to use recordLevelFilter. Hudi also uses the same config as above in `MergeOnReadIncrementalRelation`.